pkg/helpers/apply/apply.go

```diff
@@ -45,82 +47,109 @@ func ApplyDeployment(
 	client kubernetes.Interface,
 	reader asset.ScenarioReader,
 	values interface{},
+	dryRun bool,
 	headerFile string,
-	files ...string) error {
+	files ...string) ([]string, error) {
 	genericScheme.AddKnownTypes(appsv1.SchemeGroupVersion, &appsv1.Deployment{})
 	recorder := events.NewInMemoryRecorder(helpers.GetExampleHeader())
+	output := make([]string, 0)
 	//Render each file
 	for _, name := range files {
 		deploymentBytes, err := MustTempalteAsset(name, headerFile, reader, values)
 		if err != nil {
 			if IsEmptyAsset(err) {
 				continue
 			}
-			return err
+			return output, err
+		}
+		output = append(output, string(deploymentBytes))
+		if dryRun {
+			continue
 		}
 		deployment, sch, err := genericCodec.Decode(deploymentBytes, nil, nil)
 		if err != nil {
-			return fmt.Errorf("%q: %v %v", name, sch, err)
+			return output, fmt.Errorf("%q: %v %v", name, sch, err)
 		}
 		_, _, err = resourceapply.ApplyDeployment(
 			client.AppsV1(),
 			recorder,
 			deployment.(*appsv1.Deployment), 0)
 		if err != nil {
-			return fmt.Errorf("%q (%T): %v", name, deployment, err)
+			return output, fmt.Errorf("%q (%T): %v", name, deployment, err)
 		}
 	}
-	return nil
+	return output, nil
 }
 
 //ApplyDirectly applies standard kubernetes resources.
 func ApplyDirectly(clients *resourceapply.ClientHolder,
 	reader asset.ScenarioReader,
 	values interface{},
+	dryRun bool,
 	headerFile string,
-	files ...string) error {
+	files ...string) ([]string, error) {
 	recorder := events.NewInMemoryRecorder(helpers.GetExampleHeader())
+	output := make([]string, 0)
+	for _, name := range files {
```

Comment:
I would move this into a separate func such as `renderManifests`, So if dry-run is enabled, I will call it.

Reply:
ok